src/include/ikarus/utils/duneUtilities.hh

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of the Ikarus distribution (https://github.com/IkarusRepo/Ikarus).
+ * Copyright (c) 2022. The Ikarus developers.
+ *
+ * This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#pragma once
+#include <Python.h>
+
+#include <dune/fufem/dunepython.hh>
+#include <dune/functions/functionspacebases/defaultglobalbasis.hh>
+
+#include <autodiff/forward/real/real.hpp>
+namespace Ikarus {
+  template <typename... Args>
+  auto makeSharedBasis(Args&&... args) {
+    using namespace Dune::Functions::BasisFactory;
+    using DuneBasis = decltype(makeBasis(std::forward<Args>(args)...));
+    return std::make_shared<DuneBasis>(makeBasis(std::forward<Args>(args)...));
+  }
+
+  template <typename... Args>
+  auto makeConstSharedBasis(Args&&... args) {
+    using namespace Dune::Functions::BasisFactory;
+    using DuneBasis = decltype(makeBasis(std::forward<Args>(args)...));
+    return std::make_shared<const DuneBasis>(makeBasis(std::forward<Args>(args)...));
+  }
+}  // namespace Ikarus
+
+namespace Python {
```

Comment:
I am not sure but maybe this name space is already used ? can be renamed to ikarusPython ? or may be it doesn't matter here ?

Reply:
This namespace is needed since the dune-fufem function relay on the ADL (Argument dpendent lockup) which only works, if we use the correct namespace here.